From: Ubuntu Core Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Thu, 14 Feb 2019 18:59:14 -0800
Subject: setup-modules

---
 Modules/Setup | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/Modules/Setup b/Modules/Setup
index 992ffec..4a8bcd2 100644
--- a/Modules/Setup
+++ b/Modules/Setup
@@ -154,7 +154,7 @@ time timemodule.c
 #_datetime _datetimemodule.c
 # UNIVERSAL: let mpdecimal.h detect settings
 #_decimal -DUNIVERSAL -I$(srcdir)/Modules/_decimal/libmpdec _decimal/_decimal.c _decimal/libmpdec/basearith.c _decimal/libmpdec/constants.c _decimal/libmpdec/context.c _decimal/libmpdec/convolute.c _decimal/libmpdec/crt.c _decimal/libmpdec/difradix2.c _decimal/libmpdec/fnt.c _decimal/libmpdec/fourstep.c _decimal/libmpdec/io.c _decimal/libmpdec/mpalloc.c _decimal/libmpdec/mpdecimal.c _decimal/libmpdec/numbertheory.c _decimal/libmpdec/sixstep.c _decimal/libmpdec/transpose.c
-#_elementtree -I$(srcdir)/Modules/expat _elementtree.c
+#_elementtree _elementtree.c -lexpat    # elementtree accelerator
 #_heapq _heapqmodule.c
 #_json _json.c
 #_lsprof _lsprof.c rotatingtree.c
@@ -179,7 +179,7 @@ time timemodule.c
 #binascii binascii.c
 #cmath cmathmodule.c
 #math mathmodule.c
-#pyexpat -I$(srcdir)/Modules/expat  expat/xmlparse.c expat/xmlrole.c expat/xmltok.c pyexpat.c
+#pyexpat pyexpat.c -lexpat
 #unicodedata  unicodedata.c
 
 # Modules with some UNIX dependencies -- on by default:
@@ -207,7 +207,7 @@ time timemodule.c
 # Modules that require external libraries.
 
 #_bz2 _bz2module.c -lbz2
-#_ctypes _ctypes/_ctypes.c _ctypes/callbacks.c _ctypes/callproc.c _ctypes/stgdict.c _ctypes/cfield.c -ldl -lffi -DHAVE_FFI_PREP_CIF_VAR -DHAVE_FFI_PREP_CLOSURE_LOC -DHAVE_FFI_CLOSURE_ALLOC
+#_ctypes _ctypes/_ctypes.c _ctypes/callbacks.c _ctypes/callproc.c _ctypes/stgdict.c _ctypes/cfield.c _ctypes/malloc_closure.c -lffi
 # _dbm _dbmmodule.c 	# -lndbm # dbm(3)
 #_gdbm _gdbmmodule.c -lgdbm
 #_lzma _lzmamodule.c -llzma
@@ -215,6 +215,7 @@ time timemodule.c
 #_uuid _uuidmodule.c -luuid
 #zlib  zlibmodule.c -lz
 
+
 # GNU readline.  Unlike previous Python incarnations, GNU readline is
 # now incorporated in an optional module, configured in the Setup file
 # instead of by a configure script switch.  You may have to insert a
@@ -225,16 +226,8 @@ time timemodule.c
 
 #readline readline.c -lreadline -ltermcap
 
-# To dynamically link OpenSSL:
-#_ssl _ssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) $(OPENSSL_LIBS)
-#_hashlib _hashopenssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) -lcrypto
-
-# To statically link OpenSSL:
-# _ssl _ssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) \
-#     -l:libssl.a -Wl,--exclude-libs,libssl.a \
-#     -l:libcrypto.a -Wl,--exclude-libs,libcrypto.a
-# _hashlib _hashopenssl.c $(OPENSSL_INCLUDES) $(OPENSSL_LDFLAGS) \
-#     -l:libcrypto.a -Wl,--exclude-libs,libcrypto.a
+#_ssl _ssl.c -lssl -lcrypto -DPy_BUILD_CORE_BUILTIN
+#_hashlib _hashopenssl.c -lssl -lcrypto -DPy_BUILD_CORE_BUILTIN
 
 # The _tkinter module.
 #
